src/test/java/com/sivalabs/ft/features/api/controllers/FeatureControllerIntegrationTests.java

```diff
@@ -0,0 +1,431 @@
+                .uri("/api/features/{featureCode}/dependencies/{dependsOnFeatureCode}", "IDEA-1", "IDEA-2")
+                .exchange();
+        assertThat(cleanup).hasStatusOk();
+    }
+
+    @Test
+    @WithMockOAuth2User(username = "user")
+    void shouldVerifyDependencyPersistenceAcrossOperations() {
+        // Create dependency
+        var createPayload =
+                """
+            {
+                "dependsOnFeatureCode": "IDEA-2",
+                "dependencyType": "HARD",
+                "notes": "Persistence test dependency"
+            }
+            """;
+
+        var createResult = mvc.post()
+                .uri("/api/features/{featureCode}/dependencies", "IDEA-1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(createPayload)
+                .exchange();
+        assertThat(createResult).hasStatus(HttpStatus.CREATED);
+
+        // Verify dependency was persisted in database
+        verifyDependencyExists("IDEA-1", "IDEA-2", "HARD", "Persistence test dependency");
+
+        // Verify dependency exists by attempting to update it
+        var verifyExistsPayload =
+                """
+            {
+                "dependencyType": "SOFT",
+                "notes": "Updated to verify existence"
+            }
+            """;
+
+        var verifyExistsResult = mvc.put()
+                .uri("/api/features/{featureCode}/dependencies/{dependsOnFeatureCode}", "IDEA-1", "IDEA-2")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(verifyExistsPayload)
+                .exchange();
+        assertThat(verifyExistsResult).hasStatusOk();
+
+        // Verify update was persisted
+        verifyDependencyExists("IDEA-1", "IDEA-2", "SOFT", "Updated to verify existence");
+
+        // Delete dependency
+        var deleteResult = mvc.delete()
+                .uri("/api/features/{featureCode}/dependencies/{dependsOnFeatureCode}", "IDEA-1", "IDEA-2")
+                .exchange();
+        assertThat(deleteResult).hasStatusOk();
+
+        // Verify deletion was persisted in database
+        verifyDependencyNotExists("IDEA-1", "IDEA-2");
+
+        // Verify dependency no longer exists
+        var verifyDeletedResult = mvc.put()
+                .uri("/api/features/{featureCode}/dependencies/{dependsOnFeatureCode}", "IDEA-1", "IDEA-2")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(verifyExistsPayload)
+                .exchange();
+        assertThat(verifyDeletedResult).hasStatus(HttpStatus.NOT_FOUND);
+
+        // Verify deletion is persistent by trying to delete again
+        var verifyDeletePersistentResult = mvc.delete()
+                .uri("/api/features/{featureCode}/dependencies/{dependsOnFeatureCode}", "IDEA-1", "IDEA-2")
+                .exchange();
+        assertThat(verifyDeletePersistentResult).hasStatus(HttpStatus.NOT_FOUND);
+    }
+
+    // === HELPER METHODS FOR DATABASE VERIFICATION ===
```

Comment:
Lets add comment why do we need this methods. It's not clear that repo is private.